protocol.go

```diff
@@ -155,3 +157,39 @@ func (r *errorTranslatingReceiver) Receive(msg any) error {
 func (r *errorTranslatingReceiver) Close() error {
 	return r.fromWire(r.Receiver.Close())
 }
+
+func sortedAcceptPostValue(handlers []protocolHandler) string {
+	var max int
+	for _, handler := range handlers {
+		max += len(handler.ContentTypes())
+	}
+	seen := make(map[string]struct{}, max)
+	accept := make([]string, 0, max)
+	for _, handler := range handlers {
+		for contentType := range handler.ContentTypes() {
+			if _, ok := seen[contentType]; ok {
+				continue
+			}
+			seen[contentType] = struct{}{}
+			accept = append(accept, contentType)
+		}
+	}
+	sort.Strings(accept)
+	return strings.Join(accept, ", ")
+}
+
+func isCommaOrSpace(c rune) bool {
+	return c == ',' || c == ' '
+}
+
+func discard(reader io.Reader) error {
+	if lr, ok := reader.(*io.LimitedReader); ok {
+		_, err := io.Copy(io.Discard, lr)
+		return err
+	}
+	// We don't want to get stuck throwing data away forever, so limit how much
+	// we're willing to do here: at most, we'll copy 4 MiB.
+	lr := &io.LimitedReader{R: reader, N: 1024 * 1024 * 4}
```

Comment:
nit: i know this was pre-existing, but in a perfect would, this discard limit would be a const at the top of this file IMO

Reply:
I've never understood this impulse for consts that are only referenced once (why add more indirection?), but I'm in the minority. I'll change it before pushing.